src/test/java/seedu/sudohr/logic/commands/leave/ListEmployeeInLeaveCommandTest.java

```diff
@@ -0,0 +1,66 @@
+package seedu.sudohr.logic.commands.leave;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static seedu.sudohr.logic.commands.CommandTestUtil.VALID_LEAVE_DATE_LEAVE_TYPE_1;
+import static seedu.sudohr.logic.commands.CommandTestUtil.VALID_LEAVE_DATE_LEAVE_TYPE_2;
+import static seedu.sudohr.testutil.TypicalDepartments.getTypicalSudoHr;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.sudohr.commons.core.Messages;
+import seedu.sudohr.logic.commands.CommandResult;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.ModelManager;
+import seedu.sudohr.model.UserPrefs;
+import seedu.sudohr.model.leave.Leave;
+import seedu.sudohr.model.leave.LeaveDate;
+import seedu.sudohr.testutil.TypicalEmployees;
+
+public class ListEmployeeInLeaveCommandTest {
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalSudoHr(), new UserPrefs());
+        Leave leave = model
+                .getInternalLeaveIfExist(new Leave(new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))));
+        model.addEmployeeToLeave(leave, TypicalEmployees.ALICE);
+
+    }
+
+    // Handle adding null date
+    @Test
+    public void execute_listEmployeesOnNullDate_throwsCommandException() throws CommandException {
+        assertThrows(NullPointerException.class, () -> new ListEmployeeInLeaveCommand(null).execute(model));
+    }
+
+    @Test
+    public void execute_listIsFiltered_showsEverything() throws CommandException {
+        model.updateFilteredEmployeeList(unused -> false);
+        CommandResult commandResult = new ListEmployeeInLeaveCommand(new LeaveDate(LocalDate
+                .parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(model);
+        assertEquals(String.format(Messages.MESSAGE_EMPLOYEES_LISTED_OVERVIEW, 1), commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_listIsNotFiltered_showsEverything() throws CommandException {
+        // model.updateFilteredEmployeeList(unused -> true);
+        CommandResult commandResult = new ListEmployeeInLeaveCommand(new LeaveDate(LocalDate
+                .parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(model);
+        assertEquals(String.format(Messages.MESSAGE_EMPLOYEES_LISTED_OVERVIEW, 1), commandResult.getFeedbackToUser());
+    }
+
```

Comment:
just double checking on functionality, now that we swap to use of ID, it doesn't matter whether the list is filtered to show entries or not right? We can still delete a leave for an employee by referencing its id even if its entry is not shown in the current view list right?

Reply:
Can double check when testing